mino/minogrpc/server.go

```diff
@@ -386,52 +386,76 @@ type overlay struct {
 	router      router.Router
 	connMgr     session.ConnectionManager
 	addrFactory mino.AddressFactory
+	secret      interface{}
+	public      interface{}
 
 	// Keep a text marshalled value for the overlay address so that it's not
 	// calculated for each request.
 	meStr string
 }
 
-func newOverlay(me mino.Address, router router.Router,
-	addrFactory mino.AddressFactory, ctx serde.Context) (*overlay, error) {
-
-	cert, err := makeCertificate()
-	if err != nil {
-		return nil, xerrors.Errorf("failed to make certificate: %v", err)
-	}
-
-	meBytes, err := me.MarshalText()
+func newOverlay(tmpl minoTemplate) (*overlay, error) {
+	meBytes, err := tmpl.me.MarshalText()
```

Comment:
as a newcomer to the codebase, `me` is *extremely* confusing. 
Can we name this field with a more explicit name? e.g. `myAddr` or similar.